File: solutions/853. Car Fleet/Solution.py
class Solution:
    def carFleet(self, target: int, position: list[int], speed: list[int]) -> int:
        time_to_reach = [
            float(target - pos) / spd
            for pos, spd in sorted(zip(position, speed), reverse=True)
        ]
        current = result = 0
        for t in time_to_reach:
            if t > current:
                current = t
                result += 1
        return result

File: solutions/853. Car Fleet/Solution.java
import java.util.*;

class Solution {
    public int carFleet(int target, int[] position, int[] speed) {
        Map<Integer, Double> carMap = new TreeMap<>(Collections.reverseOrder());
        for (int i = 0; i < position.length; i++) {
            carMap.put(position[i], (double) (target - position[i]) / speed[i]);
        }
        int result = 0;
        double current = 0;
        for (double time : carMap.values()) {
            if (time > current) {
                current = time;
                result++;
            }
        }
        return result;
    }
}
